relay-base-schema/src/metrics/mod.rs

```diff
@@ -9,8 +9,11 @@ pub use self::name::*;
 pub use self::units::*;
 
 use regex::Regex;
+use std::cmp::min;
 use std::{borrow::Cow, sync::OnceLock};
 
+const CUSTOM_METRIC_NAME_MAX_SIZE: usize = 150;
```

Comment:
A short comment explaining the limit would be good, also can mention the technical limit in the indexer and how much space that leaves for units and namespace names.

Also we need to not forget to update the sentry dev docs.